invenio_previewer/extensions/xml_prismjs.py

```diff
@@ -24,9 +25,12 @@ def render(file):
     """Pretty print the XML file for rendering."""
     with file.open() as fp:
         encoding = detect_encoding(fp, default='utf-8')
-        file_content = fp.read().decode(encoding)
-        parsed_xml = xml.dom.minidom.parseString(file_content)
-        return parsed_xml.toprettyxml(indent='  ', newl='')
+        try:
+            file_content = fp.read().decode(encoding)
+            parsed_xml = xml.dom.minidom.parseString(file_content)
+            return parsed_xml.toprettyxml(indent='  ', newl='')
+        except UnicodeDecodeError:
+            return _("Error decoding the file. Are you sure it is " + encoding + "?");
```

Comment:
Probably this is not the correct way of marking messages that can be translated if they have a  variable inside....

Reply:
Indeed, [here](https://inveniordm.docs.cern.ch/develop/howtos/i18n/#python) how to do it.